includes/class-sensei-lesson.php

```diff
@@ -2557,23 +2561,38 @@ public function lesson_add_existing_questions() {
 
 				foreach ( $questions as $question_id ) {
 
-					++$question_count;
-
 					$quizzes = get_post_meta( $question_id, '_quiz_id', false );
+
 					if ( ! in_array( $quiz_id, $quizzes ) ) {
+
+						$at_least_one_question_added = true;
+						$number_of_added_questions++;
+						++$question_count;
 						add_post_meta( $question_id, '_quiz_id', $quiz_id, false );
 						$lesson_id = get_post_meta( $quiz_id, '_quiz_lesson', true );
 						update_post_meta( $lesson_id, '_quiz_has_questions', '1' );
-					}
-
-					add_post_meta( $question_id, '_quiz_question_order' . $quiz_id, $quiz_id . '000' . $question_count );
-					$question_type = Sensei()->question->get_question_type( $question_id );
+						add_post_meta( $question_id, '_quiz_question_order' . $quiz_id, $quiz_id . '000' . $question_count );
+						$question_type = Sensei()->question->get_question_type( $question_id );
+						$return       .= $this->quiz_panel_question( $question_type, $question_count, $question_id );
 
-					$return .= $this->quiz_panel_question( $question_type, $question_count, $question_id );
+					}
 				}
 			}
 		}
 
+		// Log event: when an existing question is added to a quiz.
+		if ( $at_least_one_question_added ) {
```

Comment:
Seems as though we can just check that `count( $questions ) > 0`.

Reply:
Please see my comment below! (In the main thread...)

Reply:
I still don't think we need both these variables though. Seems that we could do `count( $number_of_added_questions ) > 0` just as easily.

Reply:
Done!